File: backend/app/v1/schemas/job_priority.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class JobPriorityBase(BaseModel):
    duration_days: int
    associate_reminder_hours: int = 24


class JobPriorityCreate(JobPriorityBase):
    pass


class JobPriorityUpdate(BaseModel):
    duration_days: int | None = None
    associate_reminder_hours: int | None = None


class JobPriorityRead(JobPriorityBase):
    id: uuid.UUID
    name: str
    assigned_jobs_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/jobPriority.ts
import type { JobPriorityCreateFormValues, JobPriorityUpdateFormValues } from "@/schemas/jobPriority";
/**
 * Job priority returned from read operations.
 */
export interface JobPriorityRead {
  id: string;
  name: string;
  duration_days: number;
  created_at: string;
  updated_at: string;
  assigned_jobs_count: number
  associate_reminder_hours: number
}

/**
 * Payload for creating a new job priority.
 */
export type JobPriorityCreate = JobPriorityCreateFormValues

/**
 * Payload for updating an existing job priority.
 */
export type JobPriorityUpdate = JobPriorityUpdateFormValues
